packages/components/src/local/map-bar/helpers/collate-adjudication-form-data.spec.ts

```diff
@@ -0,0 +1,29 @@
+/* global it expect */
+import collateAdjudicationFormData from './collate-adjudication-form-data'
+
+import platformTypes from '@serge/mocks/platform-types.mock'
+import selectedAsset from '@serge/mocks/selected-asset.mock'
+import forces from '@serge/mocks/forces.mock'
+import { AdjudicateTurnFormPopulate, AdjudicateTurnFormValues } from '@serge/custom-types'
+
+it('contains relevant population results', () => {
+  const res: AdjudicateTurnFormPopulate = collateAdjudicationFormData(platformTypes, selectedAsset, forces).populate
+  expect(res.condition.length).toEqual(5)
```

Comment:
`const { condition, speed, status, visibleTo } = res`

would be preferred here and again in the value section.

Reply:
There are a few other examples of this in other files too.

Reply:
There's a _cost_ of adding an extra line of code.  But, can you explain the benefit?

(I recognise the above would reduce lines if 4 consts were being declared, but, it's not reducing lines in this instance)

Reply:
It's a code-style thing. Our code style is that we use destructuring, it makes code easier to read, it reduces the possibility of errors (typos for example) and in some instances (although not this one) it caches values and therefore reduces calls to external data.

Looking at that again, you could actually not bother with `res` at all and you could destructure from the `collageAdjudicationFormData` directly.

I approved the PR as it's not something you HAVE to do but if you want to stick to a consistent coding-style, then it's something you should do.

Reply:
I have dereferenced the results from the method. The TS typing bits are a bit cumbersome: https://github.com/serge-web/serge/pull/417/commits/f1dbb9938d78a22eabb5893742df591f80feb7cf#diff-dcb8674751fb00c2ffec04cd29fae785R11